unison-cli/src/Unison/Share/SyncV2.hs

```diff
@@ -645,3 +669,47 @@ withEntityLoadingCallback :: (MonadUnliftIO m) => ((Int -> m ()) -> m a) -> m a
 withEntityLoadingCallback action = do
   let msg n = "\n  📦 Unpacked  " <> tShow n <> " entities...\n\n"
   counterProgress msg action
+
+withSortedStreamProgress :: (MonadIO m, MonadUnliftIO n) => Maybe Int -> ((Int -> m (), m (), Int -> m ()) -> n a) -> n a
+withSortedStreamProgress total action = do
+  unpackedVar <- IO.newTVarIO (0 :: Int)
+  doneUnpackingVar <- IO.newTVarIO False
+  savedVar <- IO.newTVarIO (0 :: Int)
+  IO.withRunInIO \toIO -> do
+    Console.Regions.displayConsoleRegions do
+      Console.Regions.withConsoleRegion Console.Regions.Linear \region -> do
+        Console.Regions.setConsoleRegion region do
+          unpacked <- IO.readTVar unpackedVar
+          doneUnpacking <- IO.readTVar doneUnpackingVar
+          saved <- IO.readTVar savedVar
+          pure $
+            Text.unlines
+              [ "",
+                "  ⬇️ Downloaded: " <> tShow unpacked <> maybe "" (\total -> " / " <> tShow total) total <> Monoid.whenM doneUnpacking " 🏁",
```

Comment:
Is this `maybe` for a future server feature, or how do I trigger it?

Reply:
There's a slot for the server to send the total if it knows what it is, but at the moment it doesn't ever send it because it would require enumerating the entire results set before starting the stream (and would slow everything down).

But if we ever support pre-computed release bundles then the server could provide the total here :)

Reply:
Ok sounds good